lib/Doctrine/ORM/Query/AST/Functions/FunctionNode.php

```diff
@@ -43,4 +45,9 @@ public function dispatch($sqlWalker)
      * @return void
      */
     abstract public function parse(Parser $parser);
+
+    public function getReturnType(): Type
+    {
+        return Type::getType(Type::STRING);
```

Comment:
So copying my comment here: 

Why defining this when it's default fallback anyway?

Reply:
I copied this here so child classes automatically implement the TypedExpression interface.

Reply:
Why should they automatically implement it though? Then they need to override it, instead of cleanly implement it.

Reply:
This is an opinionated decision (I didn't want to specify "implements ...." on subclass). If it really bother I won't argue and fix my code according to your review

Reply:
@Grafikart I understand your decision. However, having the concrete classes implementing the interface makes things much more explicit - especially if you consider that we have many extensions providing platform-specific functions.

Please do apply the changes to have the type only on the concrete classes :+1: